runtime/lua/vim/img/_detect.lua

```diff
@@ -0,0 +1,88 @@
+local terminal = require('vim.img._terminal')
+
+local TERM_QUERY = {
+  -- Request device attributes (DA2).
+  --
+  -- It typically returns information about the terminal type and supported features.
+  --
+  -- Response format is typically something like '\033[>...;...;...c'
+  DEVICE_ATTRIBUTES = terminal.code.ESC .. '[>q',
+
+  -- Request device status report (DSR), checking if terminal is okay.
+  --
+  -- Response indicates its current state.
+  DEVICE_STATUS_REPORT = terminal.code.ESC .. '[5n',
+}
+
+local TERM_RESPONSE = {
+  -- Indicates that the terminal is functioning normally (no error).
+  --
+  -- 0 means 'OK'; other values indicate different states or errors.
+  OK = terminal.code.ESC .. '[0n',
+}
+
+---Detects supported graphics of the terminal.
+---@return {graphics:'iterm2'|'kitty'|'sixel'|nil, tmux:boolean, broken_sixel_cursor_placement:boolean}
+return function()
+  local results = { graphics = nil, tmux = false, broken_sixel_cursor_placement = false }
+
+  local term = os.getenv('TERM')
+  if term == 'xterm-kitty' or term == 'xterm-ghostty' or term == 'ghostty' then
+    results.graphics = 'kitty'
+  end
+
```

Comment:
Please DO NOT hard code `$TERM` values here. We can query for kitty graphics support and that is what we will do.

Neovim will not hard code `$TERM` values on my watch unless absolutely necessary :triumph: